lib/cosmos/tools/table_manager/table_config.rb

```diff
@@ -8,765 +8,166 @@
 # as published by the Free Software Foundation; version 3 with
 # attribution addendums as found in the LICENSE.txt
 
-require 'cosmos'
+require 'cosmos/config/config_parser'
+require 'cosmos/packets/packet_config'
 require 'cosmos/tools/table_manager/table'
+require 'cosmos/tools/table_manager/table_parser'
+require 'cosmos/tools/table_manager/table_item_parser'
 
 module Cosmos
 
-  # TableConfig provides capabilities to read an ascii file that defines the
-  # table parameters in a system and create a set of Tables for each.
-  class TableConfig
+  class TableConfig < PacketConfig
+    attr_reader :filename
 
-    # Constructor for a TableConfig
     def initialize
-      @ordered_tables = nil
-      @tables = nil
-      @current_name = nil
-      @current_table = nil
-      @current_parameter = nil
-      @cur_bit_offset = 0
-      @default_count = 0
-      @item_definitions = nil
-      @table_names = nil
+      super
+      # Override commands with the Table::TARGET name to store tables
+      @commands[Table::TARGET] = {}
     end
 
-    # Processes a file and adds in the tables defined in the file
-    def process(filename)
-      building_generic_conversion = false
-      converted_type = nil
-      converted_bit_size = nil
-      proc_text = ''
+    def tables
+      @commands[Table::TARGET]
+    end
 
-      Logger.info "Processing table config in file '#{filename}'"
+    def table_names
+      tables.keys
+    end
 
-      unless test ?f, filename
-        Logger.error "File does not exist"
-        raise "ERROR! Table config file #{filename} does not exist!"
-      end
+    def table(table_name)
+      tables[table_name.upcase]
+    end
+
+    #########################################################################
+    # The following methods process a table config file
+    #########################################################################
 
-      parser = ConfigParser.new
-      parser.parse_file(filename) do |keyword, parameters|
-        next unless keyword
-        if building_generic_conversion
+    # Processes a COSMOS table configuration file and uses the keywords to build up
+    # knowledge of the tables.
+    #
+    # @param filename [String] The name of the configuration file
+    def process_file(filename)
+      # Partial files are included into another file and thus aren't directly processed
+      return if File.basename(filename)[0] == '_' # Partials start with underscore
+      @filename = filename
+      @converted_type = nil
+      @converted_bit_size = nil
+      @proc_text = ''
+      @building_generic_conversion = false
+
+      parser = ConfigParser.new("http://cosmosrb.com/docs/tools/#table-manager-configuration")
+      parser.parse_file(filename) do |keyword, params|
+        if @building_generic_conversion
           case keyword
           # Complete a generic conversion
-          when 'GENERIC_READ_CONVERSION_END', 'GENERIC_WRITE_CONVERSION_END', 'CONSTRAINT_END'
```

Comment:
I eliminated CONSTRAINT because it generic conversions can do the exact same thing by referencing other items in the packet.